python-avd/pyavd/_cv/client/proxy.py

```diff
@@ -90,3 +102,81 @@ async def create_socket_for_grpc(self) -> socket.socket:
+            )
+        ):
+            return True
+
+        # Fallback to the proxy-related settings passed via environment variables
+        return self.proxy_set_via_env and not self.bypass_proxy
+
+    @cached_property
+    def bypass_proxy(self) -> bool:
+        """Checks if target CloudVision server is included in environment variables `no_proxy` or `NO_PROXY` forcing proxy bypass."""
+        for env_variable_candidate in ["no_proxy", "NO_PROXY"]:
+            if isinstance(no_proxy_candidate := environ.get(env_variable_candidate), str) and len(no_proxy_candidate) > 0:
+                return bool({f"{self.target_host}", f"{self.target_host + ':' + str(self.target_port)}"} & set(no_proxy_candidate.split(",")))
+        # No interesting environment variables are set
+        return False
+
+    @cached_property
+    def proxy_set_via_env(self) -> bool:
+        """
+        Verifies if usable proxy-related settings are set via either `https_proxy`, `HTTPS_PROXY`, `all_proxy` or `ALL_PROXY` environment variables.
+
+        Parses value of the `https_proxy`, `HTTPS_PROXY`, `all_proxy` and `ALL_PROXY` environment variables to obtain Proxy server parameters.
+        Order of preference: `https_proxy` else `HTTPS_PROXY` else `all_proxy` else `ALL_PROXY`.
+        Updates HTTPProxyManager if usable Proxy server is found.
+
+        Returns:
+            True if usable Proxy server has been identified via interesting environment variables, otherwise False.
+        """
+        for env_variable_candidate in ["https_proxy", "HTTPS_PROXY", "all_proxy", "ALL_PROXY"]:
+            if isinstance(proxy_candidate := environ.get(env_variable_candidate), str) and len(proxy_candidate) > 0:
+                parsed_proxy_candidate = self.parse_https_proxy_uri(proxy_candidate)
+                if self.proxy_candidate_is_usable(parsed_proxy_candidate):
+                    self.update_proxy_manager(parsed_proxy_candidate)
+                    return True
+        return False
+
+    @staticmethod
+    def parse_https_proxy_uri(https_proxy_uri: str) -> CVProxyServerCandidate:
+        """Parses string representing discovered proxy URI."""
+        urlparse_result: ParseResult = urlparse(https_proxy_uri)
+        return CVProxyServerCandidate(
+            scheme=urlparse_result.scheme,
+            host=urlparse_result.hostname,
+            port=urlparse_result.port,
+            username=urlparse_result.username,
+            password=urlparse_result.password,
+        )
+
+    @staticmethod
+    def proxy_candidate_is_usable(proxy_candidate: CVProxyServerCandidate) -> bool:
+        """Verifies that mandatory proxy settings are set in Proxy server candidate."""
+        return all(
+            [
+                proxy_candidate.scheme == "http",
+                isinstance(proxy_candidate.host, str) and len(proxy_candidate.host) > 0,
+                isinstance(proxy_candidate.port, int) and (0 < proxy_candidate.port < 65536),
```

Comment:
This is correct but the following is more explicit:
```suggestion
                isinstance(proxy_candidate.port, int) and (1 <= proxy_candidate.port <= 65535),
```

Reply:
FIxed